### DISCUSSION TITLE
fileFilter property

### DISCUSSION THREAD

**Discussion by pgalbraith** [jsonschema2dataclass/js2d-gradle#918] [Q&A]
I cannot seem to get `fileFilter` to work the way I want (or at all, for that matter).  Can anybody show me a working gradle build script that utilizes fileFilter to only process *.json files?  I was hoping something simple like this would allow me to generate classes for all the schema under the specified directory without failing on the non `*.json` files:

```
jsonSchema2Pojo {
    executions {
        main {
            io.source.setFrom files("${sourceSets.main.output.resourcesDir}/com/myproject/schema")
            io.fileFilter = { it.name.endsWith('.json') }
            kclass.targetPackage = 'com.myproject.schema'
        }
    }
}
```

... but this just leads to problems not recognizing the lambda as a FileFilter.  You can cast it to FileFilter which leads to problems with serialization.  Casting as serializable yields still more problems.  I give up.  How do I use `fileFilter`?  I've tried creating a hard implementation class of `FileFilter` in the gradle script, also no luck.

**Comment by eirnym** [comment]
Hi, 

Could you please fill me with gradle version you're using with which Java version? This is important as Groovy version used in Gradle scripts is dependent on Gradle version. 

The field type is defined as `Property<FileFilter>` to help Gradle with caching. 

In this case Groovy behind the DSL language in an example provided used has less ability to properly convert random lambda to a target type. I suggest to create an anonymous class to pass the filter.
